upgrade/1.2/Stage_3.md

```diff
@@ -1,13 +1,32 @@
 # Stage 3 - CSM Service Upgrades
 
-**IMPORTANT:**
+1. Prepare assets:
 
-> During this stage there will be a brief (approximately 5 minutes) window where pods with PVCs will not be able to migrate between nodes. This is due to a redeployment of the Ceph csi provisioners into namespaces to accommodate the newer charts and a better upgrade strategy.
+   > The install scripts will look for the RPM in `/root`, so it is important that you copy it there.
 
-Run `csm-upgrade.sh` to deploy upgraded CSM applications and services:
+   ```bash
+    ncn-m002# CSM_RELEASE=csm-1.2.0
+   ```
 
-```bash
-ncn-m002# /usr/share/doc/csm/upgrade/1.2/scripts/upgrade/csm-upgrade.sh
-```
+   - Internet Connected
+
+     ```bash
+     ncn-m002# /usr/share/doc/csm/upgrade/1.2/scripts/upgrade/prepare-assets.sh --csm-version ${CSM_RELEASE} --endpoint [ENDPOINT]
+     ```
+
+   - Air Gapped (replace the PATH_TO below with the location of the rpm)
+
+     ```bash
+     ncn-m002# /usr/share/doc/csm/upgrade/1.2/scripts/upgrade/prepare-assets.sh --csm-version ${CSM_RELEASE} --tarball-file [PATH_TO_CSM_TARBALL_FILE]
```

Comment:
If we are having them copy over the tarball in an earlier step and telling them where to put it, could we have that path here instead of having them have to go back and figure out where that was?

Reply:
Related to that...At what point are we having them copy over the tarball?   I see on this back that we have them create the directory for it and then we have them do a find in a path in that directory, but I don't see a command to have them actually copy or expand the tarball.

From Stage 2.3
```
ncn-m002# mkdir -p /etc/cray/upgrade/csm/${CSM_RELEASE}

ncn-m002# scp ncn-m001:/etc/cray/upgrade/csm/myenv /etc/cray/upgrade/csm/myenv

ncn-m002# csi_rpm=$(ssh ncn-m001 "find /etc/cray/upgrade/csm/${CSM_RELEASE}/tarball/${CSM_RELEASE}/rpm/cray/csm/ -name 'cray-site-init*.rpm'")

ncn-m002# scp ncn-m001:${csi_rpm} /tmp/cray-site-init.rpm

ncn-m002# scp ncn-m001:/root/docs-csm-*.noarch.rpm /root/docs-csm-latest.noarch.rpm


ncn-m002# rpm -Uvh --force /tmp/cray-site-init.rpm
```

Reply:
This conversation was marked resolved without answering the questions.   Setting back to unresolved.

Reply:
finally figured how to reply again, here is the commit pushed after some discussion with @johren-hpe https://github.com/Cray-HPE/docs-csm/pull/1229/commits/9dcf6b7a7e85792ce650995172ce2fc7c7caaee4